src/main/java/org/openhab/tools/analysis/checkstyle/readme/MarkdownCheck.java

```diff
@@ -94,23 +93,54 @@ private void checkReadMe(FileText fileText) {
         Set<Class<? extends Block>> enabledBlockTypes = new HashSet<>(
                 Arrays.asList(Heading.class, ListBlock.class, FencedCodeBlock.class, IndentedCodeBlock.class));
         Node readmeMarkdownNode = parseMarkdown(fileText, enabledBlockTypes);
+
         // CallBack is used in order to use the protected methods of the AbstractStaticCheck in the Visitor
         MarkdownVisitorCallback callBack = new MarkdownVisitorCallback() {
             @Override
-            public int findLineNumber(FileText fileContent, String searchedText, int startLineNumber)
-                    throws NoResultException {
-                return MarkdownCheck.this.findLineNumber(fileText, searchedText, startLineNumber);
+            public int findLineNumber(FileText fileContent, String searchedText, int startLineNumber,
+                    LineFormatterCallback lineFormatterCallback) throws NoResultException {
+                return MarkdownCheck.this.findLineNumberFormatted(fileText, searchedText, startLineNumber,
+                        lineFormatterCallback);
             }
 
             @Override
             public void log(int line, String message) {
                 MarkdownCheck.this.log(line, message);
             }
         };
-        MarkdownVisitor visitor = new MarkdownVisitor(callBack, fileText);
+        MarkdownVisitor visitor = new MarkdownVisitor(callBack, fileText, new LineFormatterCallback() {
+
+            @Override
+            public String formatLine(String line) {
+                Pattern needsEscape = Pattern.compile("\\\\" + Escaping.ESCAPABLE);
```

Comment:
But you can not be sure that all characters which have to escaped are actually escaped are you sure this will suffice for the most common or better all cases? Should we maybe determine the line number in a different way instead of trying to match it again?

Reply:
Given such changes I would expect new tests? Or maybe changed tests?

Reply:
The **Escaping.ESCAPABLE** constant is defined in the external markdown library and is used in the markdown parsing logic.  It will cover all the cases in which the markdown parser will make changes to the original text. 

The reason why I didn't include any new tests is that when I use the **verify** method that is provided by **com.puppycrawl.tools.checkstyle.BaseCheckTestSupport** and I provide no error messages to that method, it doesn't fail the test, it only logs the error. Thus, every test that I include, would pass even if it shouldn't. 

Do you have any ideas how to create tests that could fail?

Reply:
Can you please discuss it with some of your college's first